graspologic/embed/ase.py

```diff
@@ -158,4 +168,95 @@ def fit(self, graph, y=None):
+        """
+        Obtain latent positions from an adjacency matrix or matrix of out-of-sample
+        vertices. For more details on transforming out-of-sample vertices, see the
+        :ref:`tutorials <embed_tutorials>`. For mathematical background, see [2].
+
+        Parameters
+        ----------
+        X : array_like or tuple, original shape or (n_oos_vertices, n_vertices).
+
+            The original fitted matrix ("graph" in fit) or new out-of-sample data.
+            If ``X`` is the original fitted matrix, returns a matrix close to
+            ``self.fit_transform(X)``.
+
+            If ``X`` is an out-of-sample matrix, n_oos_vertices is the number
+            of new vertices, and n_vertices is the number of vertices in the
+            original graph. If tuple, graph is directed and ``X[0]`` contains
+            edges from out-of-sample vertices to in-sample vertices.
+
+        Returns
+        -------
+        array_like or tuple, shape (n_oos_vertices, n_components)
+        or (n_vertices, n_components).
+
+            Array of latent positions. Transforms the fitted matrix if it was passed
+            in.
+
+            If ``X`` is an array or tuple containing adjacency vectors corresponding to
+            new nodes, returns the estimated latent positions for the new out-of-sample
+            adjacency vectors.
+            If undirected, returns array.
+            If directed, returns ``(X_out, X_in)``, where ``X_out`` contains
+            latent positions corresponding to nodes with edges from out-of-sample
+            vertices to in-sample vertices.
+
+        Notes
+        -----
+        If the matrix was diagonally augmented (e.g., self.diag_aug was True), ``fit``
+        followed by ``transform`` will produce a slightly different matrix than
+        ``fit_transform``.
+
+        To get the original embedding, using ``fit_transform`` is recommended. In the
+        directed case, if A is the original in-sample adjacency matrix, the tuple
+        (A.T, A) will need to be passed to ``transform`` if you do not wish to use
+        ``fit_transform``.
+        """
+
+        # checks
+        check_is_fitted(self, "is_fitted_")
+        if isinstance(X, nx.classes.graph.Graph):
+            X = import_graph(X)
+        directed = self.latent_right_ is not None
+        original = directed and isinstance(X, np.ndarray)
+
+        # correct types?
+        if directed and not isinstance(X, tuple):
+            if X.shape[0] == X.shape[1]:  # in case original matrix was passed
+                msg = """A square matrix A was passed to ``transform`` in the directed case. 
+                If this was the original in-sample matrix, either use ``fit_transform`` 
+                or pass a tuple (A.T, A)."""
+                warnings.warn(msg)
+            else:
+                msg = "Directed graphs require a tuple (X_out, X_in) for out-of-sample transforms."
+                raise TypeError(msg)
```

Comment:
This might be just a github rendering prob but this format is a little off if not

Reply:
I see it was a github rendering thing! nvm!